frontend/app/utils/lookup-utils.server.ts

```diff
@@ -150,3 +150,62 @@ export function localizeAndSortPreferredLanguages(languages: Language[], locale:
     return a.name.localeCompare(b.name, locale);
   });
 }
+
+/**
+ * Localizes a single federal social program object by adding a localized name.
+ *
+ * @param program - The federal social program object to localize.
+ * @param locale - The locale code for localization.
+ * @returns The localized federal social program object with a localized name.
+ */
+export function localizeFederalSocialProgram(program: FederalSocialProgram, locale: string) {
+  const { nameEn, nameFr, ...rest } = program;
+  return {
+    ...rest,
+    name: locale === 'fr' ? nameFr : nameEn,
+  };
+}
+
+/**
+ *
+ * @param programs - The array of federal social program objects to localize and filter
+ * @param locale - The locale code for localization
+ * @param targetProgramId - The federal social program ID to filter on
+ * @returns - The localized, filtered, and joined array of federal social programs by target federal social program
+ */
+export function localizeFilterJoinFederalSocialPrograms(programs: FederalSocialProgram[], locale: string, targetProgramId: string | undefined) {
```

Comment:
I believe the filtering or search operation should be handled at the service level. Filtering by the primary key doesn't make sense here, as it will either return an empty array or an array with a single item, making the array join irrelevant.

I recommend creating a findFederalSocialProgram(id: string): FederalSocialProgram | null function that utilizes the getAllFederalSocialPrograms() function at the service level. If the result is not null, you can then call localizeFederalSocialProgram. If null is an invalid result, you can use invariant to throw an exception.

Don't forget to add `localizeFederalSocialPrograms` that will localize an array of `FederalSocialProgram` and also the `localizeAndSortFederalSocialPrograms` that will sort by locale as well. Those functions can be used in the edit screens.

Reply:
@Fbasham updated my comment ^